server/remote_cache/content_addressable_storage_server/content_addressable_storage_server.go

```diff
@@ -361,7 +363,45 @@ func (s *ContentAddressableStorageServer) fetchDir(ctx context.Context, cache in
 // Errors:
 //
 // * `NOT_FOUND`: The requested tree root is not present in the CAS.
+
+type DirectoryWithDigest struct {
+	Directory *repb.Directory
+	Digest    *repb.Digest
+}
+
+func (s *ContentAddressableStorageServer) fetchDirectory(ctx context.Context, cache interfaces.Cache, dir *repb.Directory) ([]*DirectoryWithDigest, error) {
+	if len(dir.Directories) == 0 {
+		return nil, nil
+	}
+	subdirDigests := make([]*repb.Digest, 0, len(dir.Directories))
+	for _, dirNode := range dir.Directories {
+		d := dirNode.GetDigest()
+		if d.GetHash() == digest.EmptySha256 {
+			continue
+		}
+		subdirDigests = append(subdirDigests, d)
+	}
+	rspMap, err := cache.GetMulti(ctx, subdirDigests)
+	if err != nil {
+		return nil, err
+	}
+	children := make([]*DirectoryWithDigest, 0, len(subdirDigests))
+	for _, d := range subdirDigests {
+		blob, ok := rspMap[d]
+		if !ok {
+			return nil, status.NotFoundErrorf("Digest %s not found in cache.", d.GetHash())
+		}
+		subDir := &repb.Directory{}
+		if err := proto.Unmarshal(blob, subDir); err != nil {
+			return nil, err
+		}
+		children = append(children, &DirectoryWithDigest{Directory: subDir, Digest: d})
+	}
+	return children, nil
+}
+
 func (s *ContentAddressableStorageServer) GetTree(req *repb.GetTreeRequest, stream repb.ContentAddressableStorage_GetTreeServer) error {
```

Comment:
Move the GetTree docs back here. 
BTW, it would be nice to document the GetTree implementation instead of copying the RPC documentation.

Reply:
Done. (and Done)